eternaleconomy-core/src/main/java/com/eternalcode/economy/leaderboard/LeaderboardService.java

```diff
@@ -0,0 +1,70 @@
+package com.eternalcode.economy.leaderboard;
+
+import com.eternalcode.economy.account.Account;
+import com.eternalcode.economy.account.database.AccountRepository;
+import com.eternalcode.economy.config.implementation.PluginConfig;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class LeaderboardService {
+
+    private final AccountRepository accountRepository;
+    private final PluginConfig pluginConfig;
+    private final Map<BigDecimal, List<Account>> topAccounts = new TreeMap<>(Comparator.reverseOrder());
+    private Instant lastUpdated;
+
+    public LeaderboardService(AccountRepository accountRepository, PluginConfig pluginConfig) {
+        this.accountRepository = accountRepository;
+        this.pluginConfig = pluginConfig;
+        this.loadLeaderboard();
+        this.lastUpdated = Instant.now();
```

Comment:
Better readability for me
```suggestion
        this.lastUpdated = Instant.now();
        this.loadLeaderboard();
```